src/savegame_json.cpp

```diff
@@ -1491,6 +1491,11 @@ void item::io( Archive& archive )
         // only for backward compatibility (nowadays mode is stored in item_vars)
         gun_set_mode(mode);
     }
+
+    // Fixes #16751
```

Comment:
Would be better if it described the bug, such as "Fixes bugged saves with tools containing nulls"